SpatialGDK/Source/SpatialGDK/Public/LoadBalancing/AbstractLockingPolicy.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include "GameFramework/Actor.h"
+
+#include "SpatialConstants.h"
+
+#include "AbstractLockingPolicy.generated.h"
+
+UCLASS(abstract)
+class SPATIALGDK_API UAbstractLockingPolicy : public UObject
+{
+	GENERATED_BODY()
+
+public:
+	virtual bool CanAcquireLock(const AActor* Actor) const PURE_VIRTUAL(UAbstractLockingPolicy::CanAcquireLock, return false;);
+	virtual bool IsLocked(const AActor* Actor) const PURE_VIRTUAL(UAbstractLockingPolicy::IsLocked, return false;);
+	virtual ActorLockToken AcquireLock(const AActor* Actor, FString LockName) PURE_VIRTUAL(UAbstractLockingPolicy::AcquireLock, return 0;);
```

Comment:
Should we define an `ActorLockToken InvalidActorLockToken = 0 (or -1)` and return it here instead of 0?

Reply:
Yes we should, I think it's here currently as a side effect of us being trolled by the compiler and trying to simplify what could be causing issues 😄

Reply:
I generally prefer 0. Implicitly converts to false where as -1 will convert to true. It just needs to be  sentinel value though for the most part.

Reply:
Oh wait. Misread. 
In this case it shouldn't matter as this answer is meaningless. It's a compiler error to call it. So making it a value that has a meaning might actually be worse than just returning 0.
But I don't really mind that much.

Reply:
Oh yeah, I misread too 😄 there's an invalid constant, we just forgot to add it again here

Reply:
My above comment was that it might be more correct to just return int() though. As the result isn't reall and invalid token so much as something that is needed to compile because of unreal. It's junk. So maybe -1337 would be better.
(If you want to make it the invalid number I won't object though).

Reply:
Oh right, I see the constant in `SpatialConstants.h`. So I imagine it's best to return `INVALID_ACTOR_LOCK_TOKEN` instead of `0`.